client/src/Auth/Login.js

```diff
@@ -0,0 +1,72 @@
+import { faEye } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useState } from "react";
+import { useLocation, useNavigate } from "react-router";
+import { NavLink } from "react-router-dom";
+import { useAuth } from "./auth-context";
+
+export const Login = () => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const { state } = useLocation();
+  const navigate = useNavigate();
+  const { dispatch } = useAuth();
+
+  const loginHandler = () => {
+    dispatch({
+      type: "SET_CREDENTIALS",
+      username: username,
+      password: password,
+      navigate: navigate,
+      navigateTo: state?.from,
+    });
+  };
+  return (
+    <div className='signin-form'>
+      <div className='h1'>Sign In</div>
+
+      <div>
+        <label className='text-label'>
+          Username<span className='red-color'>*</span>
+        </label>
+        <div>
+          <input
+            className='textbox'
+            type='text'
+            placeholder='Enter Username'
+            value={username}
+            onChange={(e) => setUsername(e.target.value)}
```

Comment:
alternate approach is that you can use useRef instead of e.target.value......but this also works!

Reply:
Will try that in future projects...thanks